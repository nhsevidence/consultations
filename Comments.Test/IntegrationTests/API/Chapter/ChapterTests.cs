using System;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Comments.Test.Infrastructure;
using Comments.ViewModels;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using Shouldly;
using Xunit;

namespace Comments.Test.IntegrationTests.API.Chapter
{
    public class ChapterTests : TestBase
    {
        [Theory]
        [InlineData(-1, 1, "introduction")]
        [InlineData(0, 1, "introduction")]
        [InlineData(1, 0, "introduction")]
        [InlineData(1, -1, "introduction")]
        [InlineData(1, -1, "")]
        public async Task Get_Chapter_Feed_Using_Invalid_Ids_Returns_Empty_Feed(int consultationId, int documentId, string chapterSlug)
        {
            //Arrange (in the base constructor for this one.)

            // Act
            var response = await _client.GetAsync($"/consultations/api/Chapter?consultationId={consultationId}&documentId={documentId}&chapterSlug={chapterSlug}");

            // Assert
            response.StatusCode.ShouldBe(HttpStatusCode.NotFound); //maybe should be a 500?
        }

        [Theory]
        [InlineData(null)]
        [InlineData("")]
        [InlineData("  ")]
        public async Task Get_Chapter_Feed_Using_Chapter_Slug_Throws_Error(string chapterSlug)
        {
            //Arrange (in the base constructor for this one.)

            // Act
            var response = await _client.GetAsync($"/consultations/api/Chapter?consultationId=1&documentId=2&chapterSlug={chapterSlug}");
           // response.EnsureSuccessStatusCode();
            var responseString = await response.Content.ReadAsStringAsync();

            // Assert
            //var deserialisedResponse = JsonConvert.DeserializeObject<JObject>(responseString);
            //deserialisedResponse.
            responseString.ShouldMatchApproved();
        }

        [Fact]
        public async Task Get_Chapter_Feed_Returns_Populated_Feed()
        {
            //Arrange (in the base constructor for this one.)

            // Act
            var response = await _client.GetAsync("/consultations/api/Chapter?consultationId=1&documentId=2&chapterSlug=introduction");
            response.EnsureSuccessStatusCode();
            var responseString = await response.Content.ReadAsStringAsync();

            // Assert
            responseString.ShouldMatchApproved();
        }
    }
}
