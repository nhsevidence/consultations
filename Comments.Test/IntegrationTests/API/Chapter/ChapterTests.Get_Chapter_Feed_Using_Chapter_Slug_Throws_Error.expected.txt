
<!DOCTYPE html>
  <html class="no-js" lang="en-GB">
    <head>
      <meta charset="utf-8">
        <meta content="IE=edge,chrome=1" name="X-UA-Compatible">
          <meta name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no">
            <title data-react-helmet="true">
            </title>
            <base href="/consultations/"/>
              <meta name="theme-color" content="#004650">
                <script src="/consultations/vendor/modernizr-custom.js">
                </script>
                <link rel="manifest" href="/consultations/manifest.json">
                  <meta name="msapplication-TileColor" content="#f6f6f6"/>
                  <meta name="msapplication-TileImage" content="//cdn.nice.org.uk/V3/Content/nice/favicon-144.png"/>
                    <link rel="shortcut icon" href="//cdn.nice.org.uk/V3/Content/nice/favicon.ico"/>
                      <link rel="apple-touch-icon-precomposed" href="//cdn.nice.org.uk/V3/Content/nice/favicon-152.png">
                        <link rel="apple-touch-icon" href="//cdn.nice.org.uk/V3/Content/nice/favicon-152.png">
                          <link rel="apple-touch-icon-precomposed" sizes="120x120" href="//cdn.nice.org.uk/V3/Content/nice/favicon-120.png">
                            <link rel="apple-touch-icon" sizes="120x120" href="//cdn.nice.org.uk/V3/Content/nice/favicon-120.png">
                              <link rel="apple-touch-icon-precomposed" sizes="152x152" href="//cdn.nice.org.uk/V3/Content/nice/favicon-152.png">
                                <link rel="apple-touch-icon" sizes="152x152" href="//cdn.nice.org.uk/V3/Content/nice/favicon-152.png">
                                  <link href="//fonts.googleapis.com/css?family=Lato:400,700,900" rel="stylesheet"/>
                                    <link rel="stylesheet" href="/consultations/main.css">
                                      <!--[if lt IE 9]>
                                        <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv.min.js">
                                        </script>
                                        <script src="//cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-shim.min.js">
                                        </script>
                                        <script src="//cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.10/es5-sham.min.js">
                                        </script>
                                        <script src="//cdnjs.cloudflare.com/ajax/libs/respond.js/1.4.2/respond.min.js">
                                        </script>
                                        <![endif]-->
                                        </head>
                                        <body>
                                          <div id="root">
                                            <div data-reactroot="">
                                              <div>
                                                <h1>
                                                  Not found
                                                </h1>
                                                <p>
                                                  Sorry that page could not be found.
                                                </p>
                                              </div>
                                            </div>
                                          </div>
                                          <script src="//cdn.nice.org.uk/V3.1/Scripts/nice/NICE.TopHat.min.js">
                                          </script>
                                          <script>
                                            window.__PRELOADED__={};
                                          </script>
                                          <script type="text/javascript" src="/consultations/static/js/main.js">
                                          </script>
                                        </body>
                                      </html>